### PR TITLE
Bug 1919588 - Add a new testing alert table, along with new fields.

### PR DESCRIPTION
**Pull request by gmierz**
This patch adds a new alert table for testing new detection techniques. Some things are changed with the alert table definitions to be able to reuse a lot of the code in the new tables. At the same time, one of the methods (autodetermine_status) is modified to be more generic and usable by subclasses.

Two new fields are added called `confidence`, and `detection_method`. The confidence field is a more generically termed field similar to the `t_value` field but isn't specific to T-test. The `detection_method` field will be used to store the name of the detection method used to produce the alert.

### CODE REVIEW

**Comment by gmierz** [comment]
I tested this locally on my own DB and the migrations worked without any breaking changes to the existing alert tables I had. Two new tables were created with them, and the new fields were also properly added.

Something that I'm not sure how to test though is the `autodetermine_status` changes, @beatrice-acasandrei do you know how it could be tested? I'm wondering if testing on prototype might be simplest for that.

File: treeherder/perf/models.py [hunk 1]
```diff
@@ -352,72 +352,75 @@ def update_status(self, using=None):
         self.status = self.autodetermine_status()
         self.save(using=using)
 
-    def autodetermine_status(self):
-        alerts = PerformanceAlert.objects.filter(summary=self) | PerformanceAlert.objects.filter(
+    def autodetermine_status(self, alert_model=None):
+        summary_klass = self.__class__
```
Comment (beatrice-acasandrei) [inline treeherder/perf/models.py:356]: Did you mean `summary_class`?

**Comment by beatrice-acasandrei** [comment]
> I tested this locally on my own DB and the migrations worked without any breaking changes to the existing alert tables I had. Two new tables were created with them, and the new fields were also properly added.
> 
> Something that I'm not sure how to test though is the `autodetermine_status` changes, @beatrice-acasandrei do you know how it could be tested? I'm wondering if testing on prototype might be simplest for that.



> I tested this locally on my own DB and the migrations worked without any breaking changes to the existing alert tables I had. Two new tables were created with them, and the new fields were also properly added.
> 
> Something that I'm not sure how to test though is the `autodetermine_status` changes, @beatrice-acasandrei do you know how it could be tested? I'm wondering if testing on prototype might be simplest for that.

I can do some testing locally in the following days if that helps. The only issue with prototype that I know of is that we need to let people know we introduced database changes because at the end of the testing process someone will need to revert everything if I'm remembering correctly. 

FYI since it might affect these changes, the old autodetermine status logic has some bugs and is not working correctly in some cases.

**Comment by Archaeopteryx** [comment]
> The only issue with prototype that I know of is that we need to let people know we introduced database changes because at the end of the testing process someone will need to revert everything if I'm remembering correctly.

If the database changes deployed on prototype are the same which get deployed to production, there is nothing to be done.

**Comment by gmierz** [comment]
@beatrice-acasandrei if there are some bugs in the old autodetermine_status function - I could look at fixing them if they aren't too big since I'm already here.

**Comment by beatrice-acasandrei** [comment]
> @beatrice-acasandrei if there are some bugs in the old autodetermine_status function - I could look at fixing them if they aren't too big since I'm already here.

Most recent bugs that I know of are the following:
https://mozilla-hub.atlassian.net/browse/FXP-3775
https://mozilla-hub.atlassian.net/browse/FXP-3738

**Comment by beatrice-acasandrei** [comment]
> @beatrice-acasandrei if you could do some testing on it, that would be helpful, thanks! I'm not entirely sure how to get that function to trigger locally.

I've been testing the autodetermine status changes with small summaries and it works as expected. Next I'll try with bigger summaries, perhaps I can reproduce [Bug FXP-3775](https://mozilla-hub.atlassian.net/browse/FXP-3775) and provide you more details.

**Comment by gmierz** [comment]
Thanks for the review @beatrice-acasandrei! I've resolved the bug from https://mozilla-hub.atlassian.net/browse/FXP-3738 in the last couple commits. I left a comment on the other bug since it's not related to the autodetermine_status method, but to the way in which the alert modification requests are done by the frontend.

There was a broken test, but I think the test itself is broken and not checking for the right conditions. Let me know what you think.

**Review by beatrice-acasandrei — APPROVED** [review]

**Comment by beatrice-acasandrei** [comment]
> Thanks for the review @beatrice-acasandrei! I've resolved the bug from https://mozilla-hub.atlassian.net/browse/FXP-3738 in the last couple commits. I left a comment on the other bug since it's not related to the autodetermine_status method, but to the way in which the alert modification requests are done by the frontend.
> 
> There was a broken test, but I think the test itself is broken and not checking for the right conditions. Let me know what you think.

I tested the changes locally and in my opinion we can merge the PR. Alternatively we can also push to prototype for more testing.

**Comment by gmierz** [comment]
Thanks @beatrice-acasandrei! I was able to test it locally as well and everything looked good to me so I think we can merge now.

**Pull request merged**